File: 2396-strictly-palindromic-number/Solution.js
// Return true if n is strictly palindromic and false otherwise

/**
 * @param {number} n
 * @return {boolean}
 */

var isStrictlyPalindromic = function (n) {
    for (let b = 2; b <= n - 2; b++) {
        let orig = n.toString(b);
        let rev = orig.split("").reverse().join("");

        if (orig != rev) {
            return false;
        }
    }

    return true;
};

File: 2396-strictly-palindromic-number/Solution.java
// Return true if n is strictly palindromic and false otherwise

class Solution {
    public boolean isStrictlyPalindromic(int n) {
        for (int i = 2; i <= n - 2; i++) {
            StringBuilder num = new StringBuilder(Integer.toString(n, i));

            if (num.compareTo(num.reverse()) == 0) {
                return false;
            }
        }

        return true;
    }
}
